subsys/config/src/config.c

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <errno.h>
+
+#include "mbedtls/base64.h"
```

Comment:
can't we move this base64 encoding to submodules which are actually implementing the callbacks?
This will at least allow us to add new implementations which do not depend on mbedtls.

Reply:
base64 library is called within this file (line 170, 316 calls: `mbedtls_base64_encode` and `mbedtls_base64_decode`)